Source/Serilog.Exceptions/Core/ExceptionPropertiesBag.cs

```diff
@@ -0,0 +1,46 @@
+namespace Serilog.Exceptions.Core
+{
+    using System;
+    using System.Collections.Generic;
+
+    internal class ExceptionPropertiesBag : IExceptionPropertiesBag
+    {
+        private readonly Type exceptionType;
+        private readonly IExceptionPropertyFilter filter;
+        private readonly Dictionary<string, object> properties = new Dictionary<string, object>();
+        private bool resultsCollected = false;
+
+        public ExceptionPropertiesBag(Type exceptionType, IExceptionPropertyFilter filter = null)
+        {
+            this.exceptionType = exceptionType;
+            this.filter = filter;
+        }
+
+        public IReadOnlyDictionary<string, object> ResultDictionary
+        {
+            get
+            {
+                this.resultsCollected = true;
```

Comment:
Setting this flag is kind of a side effect, I'm wondering if we should add an explicit method for this.

Reply:
Done